### PR TITLE
Puppet 4.0 support

### PR DESCRIPTION
**Pull request by ffquintella**
This a new version with some minor bugfixes

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -26,11 +26,11 @@
   "requirements": [
     {
       "name": "pe",
-      "version_requirement": ">=3.7.0 <4.0.0"
+      "version_requirement": ">=3.7.0 <4.3.0"
```
Comment (rnelson0) [inline metadata.json]: 4.3.0 was just released, is there a particular reason to stay below that version when testing?

File: .travis.yml [hunk 2]
```diff
@@ -5,17 +5,17 @@ sudo: false
...
+    env: PUPPET_GEM_VERSION="~> 3.0"
   - rvm: 2.0.0
-    env: PUPPET_GEM_VERSION="~> 3.4.0" STRICT_VARIABLES=yes
+    env: PUPPET_GEM_VERSION="~> 3.3.0" STRICT_VARIABLES=yes
   - rvm: 2.0.0
-    env: PUPPET_GEM_VERSION="~> 3.4.0" FUTURE_PARSER="yes"
+    env: PUPPET_GEM_VERSION="~> 3.0" FUTURE_PARSER="yes"
   - rvm: 2.1.0
-    env: PUPPET_GEM_VERSION="~> 3.4.0" STRICT_VARIABLES=yes FUTURE_PARSER="yes"
+    env: PUPPET_GEM_VERSION="~> 3.0" STRICT_VARIABLES=yes FUTURE_PARSER="yes"
```
Comment (rnelson0) [inline .travis.yml]: For Puppet 4 support, it seems like some 4.x gems should be in the matrix. I would suggest one entry each for ~> 4.0.0, ~> 4.1.0, ~> 4.2.0, and ~> 4.3.0, all with STRICT_VARIABLES="yes"

File: manifests/install.pp [hunk 3]
```diff
@@ -103,14 +110,13 @@
 
   if $jira::db == 'mysql' and $jira::mysql_connector_manage {
     if $jira::staging_or_deploy == 'staging' {
-      class { '::jira::mysql_connector':
+      class { 'jira::mysql_connector':
```
Comment (rnelson0) [inline manifests/install.pp]: While you no longer need colon prepends in Puppet 4, I imagine they should persist as long as Puppet 3 is supported.

File: manifests/install.pp [hunk 4]
```diff
@@ -40,10 +40,17 @@
...
+
+  if (versioncmp($jira::version, '7.0.0') < 0  ){
+    $file = "atlassian-jira-core-${jira::product}-${jira::version}${jira::format}"
+  }else {
+    $file = "atlassian-jira-core-${jira::version}${jira::format}"
+  }
   if $jira::staging_or_deploy == 'staging' {
 
-    require staging
+    class{'staging':}
```
Comment (rnelson0) [inline manifests/install.pp]: Can this be `include staging` to avoid conflicts if it's included elsewhere?

**Comment by rnelson0** [comment]
I don't have any technical comments but did add a few syntax comments. Also see https://github.com/puppet-community/puppet-jira/pull/96#issuecomment-161436990, same failures are occuring here.

File: .travis.yml (hunk 2 above)
Comment (igalic) [inline .travis.yml]: i thought this whole thing was modulesynced?

**Comment by jyaworski** [comment]
Hello:

What's the status of this PR?

File: metadata.json (hunk 1 above)
Comment (igalic) [inline metadata.json]: pretty sure we can do <5.0.0 here

**Comment by igalic** [comment]
@ffquintella can we convince you to rebase & squash this pr?

**Comment by ffquintella** [comment]
Igalic. 

For me it's ok. I will close de PR. And as soon as I have some extra time I will watch into rebasing

**Pull request closed without merging**